File: academic-system/src/main/java/org/example/academic/system/model/AcademicClass.java
package org.example.academic.system.model;

import jakarta.validation.constraints.NotBlank;
import lombok.Getter;
import lombok.ToString;

import java.util.ArrayList;
import java.util.List;

@Getter
@ToString
public class AcademicClass {

    @NotBlank(message = "Class code cannot be blank.")
    private String code;

    @NotBlank(message = "Class title cannot be blank.")
    private String title;

    private List<Assessment> assessments;

    public AcademicClass(String code, String title) {
        this.code = code;
        this.title = title;
        this.assessments = new ArrayList<>();
    }

    public void addAssessment(Assessment assessment) {
        assessments.add(assessment);
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) return true;
        if (obj == null || getClass() != obj.getClass()) return false;
        AcademicClass that = (AcademicClass) obj;
        return code.equals(that.code);
    }

    @Override
    public int hashCode() {
        return code.hashCode();
    }
}

File: academic-system-python/src/academic_system/model/academic_class.py
from __future__ import annotations
from typing import List
from academic_system.model.assessment import Assessment


class AcademicClass:
    def __init__(self, code: str, title: str):
        self.code = code
        self.title = title
        self.assessments: List[Assessment] = []

    def add_assessment(self, assessment: Assessment):
        self.assessments.append(assessment)

    def __eq__(self, other):
        if not isinstance(other, AcademicClass):
            return False
        return self.code == other.code

    def __hash__(self):
        return hash(self.code)

    def __str__(self):
        return f"AcademicClass(code={self.code}, title={self.title})"